pal/pacer/pacer-web.ts

```diff
@@ -96,32 +97,41 @@ export class Pacer {
         this._stHandle = this._stTime(updateCallback);
 
         this._isPlaying = true;
+        this._frameCount = 0;
     }
 
     stop (): void {
         if (!this._isPlaying) return;
         this._ctTime(this._stHandle);
         this._stHandle = 0;
         this._isPlaying = false;
+        this._frameCount = 0;
     }
 
     _handleRAF = () => {
-        if (performance.now() - this._start < this._delay) {
+        const elapseTime = performance.now() - this._startTime;
+        const elapseFrame = Math.floor(elapseTime / this._frameTime);
+        if (elapseFrame < this._frameCount) {
             this._rAF.call(window, this._handleRAF);
-        } else if (this._callback) {
-            this._callback();
+        } else {
+            this._frameCount++;
+            if (this._callback) {
+                this._callback();
+            }
```

Comment:
If the game logic is very complex, then `this._frameCount < elapseFrame ` is always true. In this situation, should invoke `this._callbak()` more times?

May be you can test a test case that is very complex, and set target frame to 60 to have a try.

Reply:
> If the game logic is very complex, then this._frameCount < elapseFrame  is always true. In this situation, should invoke this._callbak() more times?

As far as  this._frameCount < elapseFrame  is always true, just invoke `this._callback()` this frame and wait next frame to do next tick, wich is as expected. 

The execution frame rate will always be the same as the set frame rate (if the default is 60, it will be executed as 60). Because this way doesn't mean to make up all the frame before the delay, it just to determine whether the current frame is to be ticked or skipped. So do not need to invoke this._callbak() more times. If `large game logic` happened, I think it will return to normal in subsequent cycles over time.